### ISSUE TITLE
i am unable to reverse engineer models from my already created database 

### ISSUE THREAD

**Issue by JoeSherwood335** [dotnet/efcore#7014]
### Steps to reproduce

Sql Server 2012 
create a Custom DataType
create a Table with that Custom Datatype
run 
dotnet ef dbcontext scaffold "Datasource" "Microsoft.EntityFrameworkCore.SqlServer" -o models -t "schema.tablename" -a

### The issue
i am unable to reverse engineer models from my already created database 
using 
dotnet ef dbcontext scaffold "Datasource" "Microsoft.EntityFrameworkCore.SqlServer" -o models -t "schema.tablename" -a

```
Exception message:
no exception just a warning 
Index PK_tablename contains a column named fieldname which cannot be found on table [schema].[tablename]. Not including column in index



### Further technical details

    "Microsoft.EntityFrameworkCore.SqlServer": "1.0.1",
    "Microsoft.EntityFrameworkCore.Design":{
      "version": "1.0.0-preview2-final" ,
      "type": "build"
    },
    "Microsoft.EntityFrameworkCore.SqlServer.Design" :"1.0.1",
    "Microsoft.EntityFrameworkCore.Tools":"1.0.0-preview2-final"

Operating system: Centos 7 
Visual Studio version: VS Code

**Comment by rowanmiller** [comment]
Can you share the create table statement for your table. This is probably somewhat by-design in that we don't support custom data types, but we would expect a warning that we couldn't map the column before a warning about the PK.

**Comment by JoeSherwood335** [comment]
CREATE TYPE [dbo].[FieldType] FROM [varchar](6) NOT NULL

CREATE TABLE [schema].[table](
	[FieldName] [dbo].[FieldType] NOT NULL,
        [FieldName2] [nvarchar](11) NOT NULL

 CONSTRAINT [PK_tk] PRIMARY KEY CLUSTERED 
(
	[FieldName] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]

**Comment by lajones** [comment]
Could you try running the command with the `--verbose` flag turned on and post the results please?

**Comment by JoeSherwood335** [comment]
this is the table def i am using to create the model 
i am using Field Id as my primary key with DataType Custom
```
CREATE TABLE [dbo].[TableName2](
	[Id] [dbo].[RegistrarNo] NOT NULL,
	[FieldName2] [varchar](50) NOT NULL,
 CONSTRAINT [PK_TableName2] PRIMARY KEY CLUSTERED 
(
	[Id] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]

```

here is a portion of log, i took out anything that didnt have TableName2.

```
Found table with schema: dbo, name: TableName2.
Found column with schema: dbo, table: TableName2, colu
Found index column with schema: dbo, table: TableName2, index name: PK_TableName2, unique: True, type description: CLUSTERED, column name: Id, ordinal: 1.
Index PK_TableName2 contains a column named Id which cannot be found on table [dbo].[TableName2]. Not including column in index.
Unable to identify the primary key for table 'dbo.TableName2'.
Unable to generate entity type for table 'dbo.TableName2'.


```
it looks like to me field [Id] is completely ignored and because it's the primary key of the table.   its prevents the model from being created

**Comment by lajones** [comment]
Note for triage: this sounds like a regression. At the very least the `--verbose` flag ought to output a line that says:

```
Found column with schema: dbo, table: TableName2, column name: Id, data type: RegistrarNo ...
```

potentially with a message saying that it doesn't understand that data type (but see later).

Once it fails to see that column the rest of the errors make sense - we don't have a PK and we can't create an Entity which doesn't have a PK. 

But when last I looked we did understand type aliases - which is what this appears to be - so we should investigate further to understand why the code a) is not seeing this column at all, perhaps because b) it is not understanding the type alias.

**Comment by lajones** [comment]
Testing with VS 15.0.0+26228.0.d15rel updated with the fix from #7315. With that fix and the `--verbose` flag on I do see the messages expected above. So the regression was fixed.

I am unable to repro the problem about not understanding the particular data type mentioned above when it is defined in schema `dbo` (but see #7576 for type aliases with the same name defined in different schemas).

Also note: the above defines the type alias as based on `varchar(6)`. If I define it as based on `sql_variant`, that is a type which is not understood by our `TypeMapper`. In that case I see warning and info messages saying it could not find the type mapping for that data type and hence it skips the column and hence a PK could not be scaffolded and hence the entity can't be created. That is all as expected - so closing this as no repro.

But @JoeSherwood335 feel free to try out by downloading from the [nightly feed](https://dotnet.myget.org/F/aspnetcore-ci-dev/api/v3/index.json) (make sure you have the latest `Microsoft.EntityFrameworkCore.Design` and `Microsoft.EntityFrameworkCore.SqlServer.Design` packages installed and not just the `1.1.1` ones) and reopen if you still experience the same problem.
